h2o-py/h2o/model/model_base.py

```diff
@@ -539,6 +539,46 @@ def scoring_history(self):
         if "scoring_history" in model and model["scoring_history"] is not None:
             return model["scoring_history"].as_data_frame()
         print("No score history for this model")
+        
+    def negative_log_likelihood(self):
+        """
+        Retrieve model negative likelihood function value from scoring history if exists for GLM model
+        
+        :return: the negative likelihood function value
+        """
+        return self.extract_scoring_history("negative_log_likelihood")
+
+    def avarege_objective(self):
+        """
+        Retrieve model average objective function value from scoring history if exists for GLM model.  If there is no 
+        regularization, the avearge objective value*obj_reg should equal the neg_log_likelihood value.
+        
+        :return: the average objective function value
+        """
+        return self.extract_scoring_history("objective")
+
+        
+    def extract_scoring_history(self, value):
+        model = self._model_json["output"]
+        if 'glm' == self.algo:
+            if self.actual_params['generate_scoring_history'] is True:
+                if "scoring_history" in model and model["scoring_history"] is not None:
+                    sc_history = model["scoring_history"]
+                    col_header = sc_history._col_header
+                    if value in col_header:
+                        index_val = col_header.index(value)
+                        scLen = len(sc_history._cell_values)
+                        return sc_history._cell_values[scLen-1][index_val]
+                    else:
+                        print("{0} not available.".format(value))
+                else:
+                    raise H2OValueError("GLM scoring_history is missing.  Cannot extract {0}.".value)
+            else:
+                raise H2OValueError("For now, need to set generate_scoring_history to True to get {0}".format(value))
+        else:
+            raise H2OValueError("{0} is only for GLM models.".format(value))
```

Comment:
I don't understand why this should be limited just for glm. The scoring history is `H2OTwoDimTable` for all the models so this method could be useful for other models as well.

If you want to keep it just for GLM, I'd recommend to implement it as an extension - create a mixin class in `h2o-py/h2o/model/extensions` and then register it in `h2o-bindings/bin/custom/python/gen_glm.py` https://github.com/h2oai/h2o-3/blob/a81177da044cae1cc340fbcba469c11eaff9b9e9/h2o-bindings/bin/custom/python/gen_glm.py#L1-L7

Reply:
I agree, I don't want to limit this function to just GLM.  However for now I am doing this as a quick fix.  I have opened an JIRA to resolve this issue at least for GLM but I think you are right, this needs to be added to more models.

Reply:
The loglikelihood and objective functions should have been part of the model metrics and not for us to get it from scoring_history at all.  The two values are evaluated during the scoring part anyway, we can just add it to the list of model metrics. In the long run, I do not want to get these two values out of the scoring history at all.

Reply:
Then for the quick fix, I'd suggest to make this method "private" (prefix with underscore) for now. (Now it's added to `model_base` so user of any model could be tempted to use it.)
In R, you don't export this function as well.

Reply:
true.  Done.